source/wifi/wifi_hal.c

```diff
@@ -3706,14 +3705,16 @@ INT wifi_getApWpaEncryptionMode(INT apIndex, CHAR *output_string)
         snprintf(output_string, 32, "None");
         return RETURN_OK;
     }
-
-    if((strcmp(buf,"3")==0))
-        params.name = "rsn_pairwise";
-    else if((strcmp(buf,"2")==0))
-        params.name = "rsn_pairwise";
+    else if((strcmp(buf,"3")==0) || (strcmp(buf,"2")==0))
+    {
+        params.name = malloc(strlen("rsn_pairwise"));
```

Comment:
free is not called for this allocation. can simple assign _rsn_pairwise_ to _params.name_